### PR TITLE
[Bug][Mirror] Show type of Hidden Power or other variable type moves in battle and summary

### PR DESCRIPTION
**Pull request by flx-sta**
> Mirror of #1330 

## What are the changes?

Some move types vary depending on the user or other factors

## What did change?

This is now checked and applied to the UI (also damage calc? idk, I just copied)

### Screenshots/Videos

see #1330 

https://github.com/user-attachments/assets/0bef9e3d-71ed-4782-b532-1bb0f17c8036

## How to test the changes?

see #1330

## Checklist
- [x] **I'm using `beta` as my base branch**
- [ ] There is no overlap with another PR? 
    - => #1330 
- [ ] The PR is self-contained and cannot be split into smaller PRs?
    - =>  there is the rename of ~~MoveTypeChangeAbAttr~~ => MoveTypeChange**Ab**Attr
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue? 
    - => no, I just want to mirror
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -119,6 +118,8 @@ export default class Move implements Localizable {
   private conditions: MoveCondition[];
   private flags: integer;
   private nameAppend: string;
+  /** Used for possible type change due ot outside effects */
```
Comment (torranx) [inline src/data/move.ts:121]:
```suggestion
  /** Used for possible type change due to outside effects */
```

**Comment by innerthunder** [comment]
There's some overlap with #3643 here

**Comment by Xavion3** [comment]
I really don't like this storing data on the global move instance shared by all pokemon, this seems incredibly likely to result in some obscure edge case bugging out and leaking data between pokemon.

**Comment by frutescens** [comment]
Do you know if your fix could resolve the issue here? https://github.com/pagefaultgames/pokerogue/issues/3570

**Comment by flx-sta** [comment]
Probably not needed anymore

**Pull request closed without merging**